src/client/options/mod.rs

```diff
@@ -304,6 +304,9 @@ pub struct ClientOptions {
     #[builder(default)]
     pub(crate) zlib_compression: Option<i32>,
 
+    #[builder(default)]
+    original_srv_hostname: Option<String>,
```

Comment:
Although we already stored the original URI, it's more convenient to store the actual single hostname that we know exists in the case of SRV due to the rules about not having multiple hosts or a port.